src/transformStateWithClones.js

```diff
@@ -6,8 +6,38 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const clones = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    const copyOfState = { ...currentState };
+
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(copyOfState, action.extraData);
+        break;
+
+      case 'removeProperties':
```

Comment:
Consider will be better if you create a variables for these words and use them everywhere